EricGray/homework.js

```diff
@@ -0,0 +1,386 @@
+//Part 1: A Cards Toolkit!>
+
+
+
+var suitARR = [" Hearts", " Diamonds", " Spades", " Clubs"];
+var rankARR = ["Ace ", "Two ", "Three ", "Four ", "Five ", "Six ", "Seven ", 
+"Eight ", "Nine ", "Ten ", "Jack ", "Queen ", "King "];
+
+function checkValidID(id) {
+	if (id < 0 || id > 51 || id%1 !== 0) {
+		alert("Sorry, your number is out of range. Only whole numbers 0-51, please!)");
+		return NaN;
+			} else if (typeof id === "string" || typeof id === "boolean"
+			 || typeof id === "undefined") {
+		alert("Sorry, only whole numbers 0-51, please!");
+		return NaN;
+	} else {
+		return id;
+	} 
+;}
+
+var cardTools = { 
+
+rank: function(id) { // --> 1..13
+		if (id < 0 || id > 51 || id%1 !== 0) {
+		alert("Sorry, your number is out of range. Only whole numbers 0-51, please!)");
+		return NaN;
+			} else if (typeof id === "string" || typeof id === "boolean"
+			 || typeof id === "undefined") {
+		alert("Sorry, only whole numbers 0-51, please!");
+		return NaN;
+	}; 
+	var r = id / 4;
+	return Math.floor(r + 1);
+},
+//}
+
+suit: function(id) { // --> 1..4
+			if (id < 0 || id > 51 || id%1 !== 0) {
+		alert("Sorry, your number is out of range. Only whole numbers 0-51, please!)");
+		return NaN;
+			} else if (typeof id === "string" || typeof id === "boolean"
+			 || typeof id === "undefined") {
+		alert("Sorry, only whole numbers 0-51, please!");
+		return NaN;
+	};
+	var s = (id%4)+1;
+	return s
+},
+
+cardID: function(rank,suit) { // --> 0..51
+		if (rank <= 0 || rank > 13 || rank%1 !== 0) {
+	alert("Sorry, your number is out of range. Only whole numbers 0-13, please!)");
+	return NaN;
+		} else if (typeof rank === "string" || typeof rank === "boolean"
+		|| typeof rank === "undefined") {
+	alert("Sorry, only whole numbers 0-13, please!");
+	return NaN;
+};
+		if (suit <= 0 || suit > 4 || suit%1 !== 0) {
+	alert("Sorry, your number is out of range. Only whole numbers 0-4, please!)");
+	return NaN;
+		} else if (typeof suit === "string" || typeof suit === "boolean"
+		|| typeof suit === "undefined" || typeof suit === undefined) {
+	alert("Sorry, only whole numbers 0-4, please!");
+	return NaN;
+};
+	var cid = ((rank * 4) - 5)  + suit;
+	return cid;
+},
+
```

Comment:
I understand that you probably wrote the validation tests inside this function in order to customize the message for "0-4 whole numbers" but you could have also pulled that functionality into your is valid test above like so:

```
isValid: function(num,low,high) { // Returns--> NaN, true
        if ((typeof num)!="number") //wrong type
            return NaN;
        if (num%1 !== 0) //non-integer
            return NaN;
        if (num<low || num>high) //out of range
            return NaN;
        return true;
    },
```